nexus/src/app/deployment.rs

```diff
@@ -0,0 +1,344 @@
+            authz::FLEET,
+            blueprint_id,
+            LookupType::ById(blueprint_id),
+        );
+        opctx.authorize(Action::Delete, &blueprint).await?;
+
+        let mut blueprints = self.blueprints.lock().unwrap();
+        if let Some(target_id) = blueprints.target.target_id {
+            if target_id == blueprint_id {
+                return Err(Error::conflict(format!(
+                    "blueprint {} is the current target and cannot be deleted",
+                    blueprint_id
+                )));
+            }
+        }
+
+        if blueprints.all_blueprints.remove(&blueprint_id).is_none() {
+            return Err(blueprint.not_found());
+        }
+
+        Ok(())
+    }
+
+    pub async fn blueprint_target_view(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<BlueprintTarget, Error> {
+        opctx.authorize(Action::Read, &authz::BLUEPRINT_CONFIG).await?;
+        let blueprints = self.blueprints.lock().unwrap();
+        Ok(blueprints.target.clone())
+    }
+
+    pub async fn blueprint_target_set(
+        &self,
+        opctx: &OpContext,
+        params: BlueprintTargetSet,
+    ) -> Result<BlueprintTarget, Error> {
+        opctx.authorize(Action::Modify, &authz::BLUEPRINT_CONFIG).await?;
+        let new_target_id = params.target_id;
+        let enabled = params.enabled;
+        let mut blueprints = self.blueprints.lock().unwrap();
+        if let Some(blueprint) = blueprints.all_blueprints.get(&new_target_id) {
+            if blueprint.parent_blueprint_id != blueprints.target.target_id {
+                return Err(Error::conflict(&format!(
+                    "blueprint {:?}: parent is {:?}, which is not the current \
+                    target {:?}",
+                    new_target_id,
+                    blueprint
+                        .parent_blueprint_id
+                        .map(|p| p.to_string())
+                        .unwrap_or_else(|| String::from("<none>")),
+                    blueprints
+                        .target
+                        .target_id
+                        .map(|p| p.to_string())
+                        .unwrap_or_else(|| String::from("<none>")),
+                )));
+            }
+            blueprints.target = BlueprintTarget {
+                target_id: Some(new_target_id),
+                enabled,
+                time_set: chrono::Utc::now(),
+            };
+
+            // When we add a background task executing the target blueprint,
+            // this is the point where we'd signal it to update its target.
+            Ok(blueprints.target.clone())
+        } else {
+            Err(Error::not_found_by_id(ResourceType::Blueprint, &new_target_id))
+        }
+    }
+
+    async fn blueprint_planning_context(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<PlanningContext, Error> {
+        let creator = self.id.to_string();
+        let datastore = self.datastore();
+        let collection = datastore
```

Comment:
Yeah, I avoided thinking too hard about this because it seems like a can of worms and I figured we'd know more about what errors are ambient and problematic vs. not once we had deployed inventory in the wild.  Maybe that point is actually now, but I think what I'd like to do for now is to make the caller (which is basically the person running omdb at this point) specify the collection, leaving it their responsibility to make sure it's okay.  If/when we automate this, we can revisit based on what we found.  Does that sound okay?

Reply:
I want ahead and implemented this in 06825c065.

Reply:
Yeah, that seems like a nicer workaround. "I the caller have told you to explicitly use collection X, so presumably I am okay with any errors it reports" feels much more robust than "generate from the latest collection (that might be newer than the last one I saw) and ignore any errors it has".